### PR TITLE
Jbuilder dune rename

### PR DESCRIPTION
**Pull request by tuncer**
As discussed with @kit-ty-kate.

### CODE REVIEW

**Comment by tuncer** [comment]
The motivation is that dune provides `dune` plus `jbuilder`, and therefore depending on the package named `jbuilder` is not the right thing anymore. This doesn't mean projects will be automatically migrated to `dune`. Rather, it's a fix for the post-install message. So, projects will still use jbuilder until they're adapted to the dune changes. https://github.com/ocaml/opam-repository/blob/d8d70fb664f9fac3f64807704d6817bbfb2de06f/packages/jbuilder/jbuilder.transition/opam#L14-L15

**Comment by tuncer** [comment]
5e0e5d9?

**Comment by kit-ty-kate** [comment]
mmh I'm not sure to follow? Why remove jbuilder constraints when they are relevant? Only the one that you switched to dune are not relevant anymore

**Comment by tuncer** [comment]
That makes more sense. One moment.

**Comment by tuncer** [comment]
Done.

File: packages/camltc/camltc.0.9.8/opam [hunk 1]
```diff
@@ -11,7 +11,7 @@ build: [
 ]
 depends: [
   "ocaml" {> "4.02.3"}
-  "dune" {build & >= "1.1.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/camltc/camltc.0.9.8/opam]: This change is wrong :/

File: packages/cfg/cfg.2.2.0/opam [hunk 2]
```diff
@@ -14,7 +14,7 @@ build: [
 
 depends: [
   "ocaml" {>= "4.04"}
-  "dune" {build & >= "1.1.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/cfg/cfg.2.2.0/opam]: Same comment

File: packages/cohttp-lwt-jsoo/cohttp-lwt-jsoo.1.1.1/opam [hunk 3]
```diff
@@ -15,7 +15,7 @@ homepage: "https://github.com/mirage/ocaml-cohttp"
 bug-reports: "https://github.com/mirage/ocaml-cohttp/issues"
 depends: [
   "ocaml" {>= "4.03.0"}
-  "dune" {build & >= "1.1.0"}
+  "dune" {build}
```
Comment (kit-ty-kate) [inline packages/cohttp-lwt-jsoo/cohttp-lwt-jsoo.1.1.1/opam]: same

**Comment by kit-ty-kate** [comment]
I'm not going to go through every dune constraints removed because it would take too much time and it would be redundant but they should be let as is.

**Comment by kit-ty-kate** [comment]
Anybody can comment on this? @rgrinberg @diml ?

**Comment by kit-ty-kate** [comment]
Okay, good. I reviewed everything, fixed some mistakes and I'm just gonna wait for Camelus to tell us it doesn't break anything and move forward with the merge.

**Comment by kit-ty-kate** [comment]
`these releases are not installable anymore (78)` mmh :/

**Comment by tuncer** [comment]
What do we do?

**Comment by kit-ty-kate** [comment]
> What do we do?

Mmh I don't know. This makes a lot of packages (even if they are mostly old versions) not installable anymore, and it's annoying. We can try to see where it comes from and maybe try to fix it but I'm not sure it's feasible.

**Comment by avsm** [comment]
This PR seems to be heading in the right direction. Why not simply leave the `jbuilder` dependency on the packages that are uninstallable (since they require an older beta of jbuilder) and keep the rest?  Once you get to a set of diffs that have no uninstallable packages it should be good to merge!

**Comment by avsm** [comment]
I'm closing this one as it's causing some epic CI churn.  Please do reopen when it's appropriate to merge it.   I am pro merging this with just the installable packages included.

**Pull request closed without merging**